src/lpython/semantics/python_comptime_eval.h

```diff
@@ -131,12 +131,13 @@ struct PythonIntrinsicProcedures {
     }
 
     static ASR::expr_t *eval_str(Allocator &al, const Location &loc, Vec<ASR::expr_t*> &args) {
+        LFORTRAN_ASSERT(ASRUtils::all_args_evaluated(args));
```

Comment:
This assert might stay --- once we allow unevaluated arguments that we seem to need for some intrinsic functions, then the callback can sometimes have arguments which are not evaluated. So having this asssert statement there is a good idea, to catch bugs early.

Reply:
I made a mistake, I thought this was removing the assert. Yes, adding the asserts is a good idea.